core/validatorapi/router.go

```diff
@@ -908,30 +913,13 @@ func proxyHandler(eth2Cl eth2wrap.Client) http.HandlerFunc {
 		proxy.ErrorLog = stdlog.New(io.Discard, "", 0)
 
 		defer observeAPILatency("proxy")()
-		proxy.ServeHTTP(proxyResponseWriter{w.(writeFlusher)}, r)
+		proxy.ServeHTTP(proxyResponseWriter{w.(writeFlusher)}, r.Clone(ctx))
 	}
 }
 
 // getBeaconNodeAddress returns an active beacon node proxy target address.
-func getBeaconNodeAddress(ctx context.Context, eth2Cl eth2wrap.Client) (*url.URL, error) {
+func getBeaconNodeAddress(eth2Cl addressProvider) (*url.URL, error) {
 	addr := eth2Cl.Address()
-	if addr == "none" {
```

Comment:
this logic was only applicable to go-eth2-client's "multi" package, which we don't use anymore